src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -1001,15 +1002,17 @@ internal static Expression ConvertValue(Expression expr, List<AttributeBaseAst>
         internal static RuntimeDefinedParameterDictionary GetParameterMetaData(ReadOnlyCollection<ParameterAst> parameters, bool automaticPositions, ref bool usesCmdletBinding)
         {
             var md = new RuntimeDefinedParameterDictionary();
-            var listMd = new List<RuntimeDefinedParameter>();
+            var listMd = new List<RuntimeDefinedParameter>(parameters.Count);
```

Comment:
`listMd` isn't immediately clear as a name to me, although maybe in the context of this method it is. Just a note

Reply:
Will change the name in a separate commit.